### PR TITLE
gh-96143: Add some comments and minor fixes missed in the original PR

### PR DESCRIPTION
**Pull request by pablogsal**
Automerge-Triggered-By: GH:pablogsal

* Issue: gh-96143

### CODE REVIEW

File: Doc/using/cmdline.rst [hunk 1]
```diff
@@ -582,6 +582,8 @@ Miscellaneous options
    .. versionadded:: 3.11
       The ``-X frozen_modules`` option.
 
+   .. versionadded:: 3.11
```
Comment (kumaraditya303) [inline Doc/using/cmdline.rst]:
```suggestion
   .. versionadded:: 3.12
```

Unless you want to backport to 3.11 😆

Comment (pablogsal) [reply Doc/using/cmdline.rst]: Whoops 😉

File: Objects/perf_trampoline.c [hunk 2]
```diff
@@ -284,12 +284,23 @@ new_code_arena(void)
...
+    // measurable performance improvement by rounding trampolines up to 32-bit
+    // or 64-bit alignment.
 
     size_t n_copies = mem_size / code_size;
     for (size_t i = 0; i < n_copies; i++) {
         memcpy(memory + i * code_size, start, code_size * sizeof(char));
     }
     // Some systems may prevent us from creating executable code on the fly.
+    // TODO: Call icache invalidation intrinsics if available:
+    // __builtin___clear_cache/__clear_cache (depending if clang/gcc). This is
```
Comment (kumaraditya303) [inline Objects/perf_trampoline.c:301]: The kernel would clear the icache for us. Not sure if this would ever be a problem.

Comment (pablogsal) [reply Objects/perf_trampoline.c:301]: That's not correct in the general case. You are supposed to call this in aarch64 *if* you modify the code in some executable segment after it was read at least once. This doesn't apply for us but is not true that in all systems the kernel will clear it for us

Comment (pablogsal) [reply Objects/perf_trampoline.c:301]: See the discussion in the previous pr

**Pull request merged**